vpype/write.py

```diff
@@ -177,6 +177,40 @@ def write_hpgl(vector_data: VectorData, output, size: Tuple[float, float]) -> No
     :param output: file object to write to
     :param size: size of the page in pixel
     """
+    # for plotters A2 and above we need to offset the coords (LL = -309, -210)
+    offset = [-309, -210]
```

Comment:
I suppose we will need to deal with this based on the `--page-format` parameter...

Reply:
The most logical way is... 

if < A2 then offset = [0,0]
if > A3 then offset = [pagewidth/2, pageheight/2]

Reply:
Maybe something like:

```
if width[0] > PAGE_FORMATS["a3"][0] or width[1] > PAGE_FORMATS["a3"][1]:
   offset = (width[0]/2, width[1]/2)
else:
   offset = (0, 0)
```

This leads to a question about how to deal with `--landscape`. Does it make sense to "load" the paper in landscape instead of in portrait? If not, then we need to keep `width` as in portrait orientation and rotate geometries by 90 degree before centring. This will require work in the main `write` function.

Reply:
With my A2 plotter... I would always "load" A3 paper landscape.

Reply:
OK. I assume you could load it in portrait as well without issue right? We really must ensure that the thing behaves "as expected". So I think we must assume that presence of `--landscape` corresponds to paper loading.

Reply:
yes, landscape would also be default (AFAIK HPGL defaults to landscape)

Reply:
Actually I'm mistaken. `--landscape` should control the final appearance, ie if the graphics are upright in portrait or landscape orientation. We might consider an HPGL specific option about paper loading, maybe `--portrait-loading` (I'm not too fan, there must be something better.

Assuming the default of landscape paper loading but portrait rendering (so the graphics should exit the printer "sideways"), does the plotter automatically rotate the geometries or not?

Reply:
I think the plotter has a mode for auto — but I never trust it or use it. Much prefer drawing stuff in landscape to begin with. With a flatbed plotter... things are just that way up.